contracts/DCAStrategies/DCAStrategies/DCAStrategiesPositionsHandler.sol

```diff
@@ -82,7 +83,32 @@ abstract contract DCAStrategiesPositionsHandler is IDCAStrategiesPositionsHandle
     uint256 _positionId,
     uint256 _amount,
     uint32 _newSwaps
-  ) external override {}
+  ) external onlyWithPermission(_positionId, IDCAStrategies.Permission.INCREASE) {
+    Position memory _position = _userPositions[_positionId];
+    IDCAStrategies.ShareOfToken[] memory _tokens = _getTokenShares(_position.strategyId, _position.strategyVersion);
+
+    // extract money from user
+    IERC20(_position.fromToken).safeTransferFrom(msg.sender, address(this), _amount);
```

Comment:
There is a valid case where `_amount` might be 0. This could happen when the user has some remaining funds and they want to re-distribute them in a new amounts of swaps. When that happens, it would be cheaper to avoid calling `safeTransferFrom` and `_approveHub`